relayer/store/beefy.go

```diff
@@ -0,0 +1,141 @@
+package store
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	merkletree "github.com/wealdtech/go-merkletree"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/sha3"
+)
+
+type NewSignatureCommitmentMessage struct {
+	Payload                       [32]byte
+	ValidatorClaimsBitfield       []*big.Int
+	ValidatorSignatureCommitment  []byte
+	ValidatorPosition             *big.Int
+	ValidatorPublicKey            common.Address
+	ValidatorPublicKeyMerkleProof [][32]byte
+}
+
+type CompleteSignatureCommitmentMessage struct {
+	ID                               *big.Int
+	Payload                          [32]byte
+	RandomSignatureCommitments       [][]byte
+	RandomSignatureBitfieldPositions []*big.Int
+	RandomValidatorAddresses         []common.Address
+	RandomPublicKeyMerkleProofs      [][][32]byte
+}
+
+type Beefy struct {
```

Comment:
I think this is a BeefyWitness right?

Reply:
Yea it's a witnessed Beefy justification that includes the signed commitment + authority addresses. `BeefyWitness` sounds like it refers to the party running the relayer that 'witnessed' the Beefy justification. How about `WitnessedBeefyJustification` or `BeefyJustification`?

Reply:
https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/05ed130b01538ab8198d39a72e0e594352c4ae24